src/pages/user-rules.js

```diff
@@ -0,0 +1,270 @@
+import React, { useEffect, useState } from 'react';
+
+import LatestRulesContent from '../components/latest-rules-content/latestRulesContent';
+import Breadcrumb from '../components/breadcrumb/breadcrumb';
+import SideBar from '../components/side-bar/side-bar';
+import { graphql } from 'gatsby';
+import { objectOf } from 'prop-types';
+import qs from 'query-string';
+import { FilterOptions } from '../components/filter/filter';
+import { ApplicationInsights } from '@microsoft/applicationinsights-web';
+const appInsights = new ApplicationInsights({
+  config: {
+    instrumentationKey: process.env.APPINSIGHTS_INSTRUMENTATIONKEY,
+  },
+});
+
+const ActionTypes = {
+  BEFORE: 'before',
+  AFTER: 'after',
+};
+
+const UserRules = ({ data, location }) => {
+  const [notFound, setNotFound] = useState(false);
+  const [filteredItems, setFilteredItems] = useState({ list: [], filter: {} });
+  const [isAscending, setIsAscending] = useState(true);
+  const [previousPageCursor, setPreviousPageCursor] = useState([]);
+  const [nextPageCursor, setNextPageCursor] = useState('');
+  const [tempCursor, setTempCursor] = useState('');
+  const [hasPrevious, setHasPrevious] = useState(false);
+  const [hasNext, setHasNext] = useState(false);
+
+  const filterTitle = 'Results';
+  const rules = data.allMarkdownRemark.nodes;
+
+  const queryStringSearch = qs.parse(location?.search, {
+    parseNumbers: true,
+  });
+
+  const queryStringRulesAuthor = queryStringSearch.author || '';
+
+  useEffect(() => {
+    fetchPageData();
+  }, []);
+
+  const fetchPageData = async (action) => {
+    try {
+      const searchData = await fetchGithubData(action);
+      const resultList = searchData.nodes;
+      const extractedFiles = getExtractedFiles(resultList);
+      const filteredRules = filterUniqueRules(extractedFiles);
+      updateFilteredItems(filteredRules);
+    } catch (err) {
+      setNotFound(true);
+      appInsights.trackException({
+        error: new Error(err),
+        severityLevel: 3,
+      });
+    }
+  };
+
+  const fetchGithubData = async (action) => {
+    const githubOwner = process.env.GITHUB_ORG;
+    const githubRepo = process.env.GITHUB_REPO;
+    const token = process.env.GITHUB_API_PAT;
+    const apiBaseUrl = 'https://api.github.com/graphql';
+
+    let cursorQuery = '';
+    if (action === ActionTypes.BEFORE) {
+      cursorQuery = `before: "${
+        previousPageCursor[previousPageCursor.length - 1]
+      }"`;
+    } else if (action === ActionTypes.AFTER) {
+      setPreviousPageCursor([...previousPageCursor, tempCursor]);
+      cursorQuery = `after: "${nextPageCursor}"`;
+    }
+
+    const query = `
+        {
+          search( query: "repo:${githubOwner}/${githubRepo} is:pr base:${process.env.CONTENT_BRANCH} is:merged sort:updated-desc author:${queryStringRulesAuthor}"
+          type: ISSUE
+          first: 30
```

Comment:
Hardcoded values in queries and URLs (e.g., first: 30, first: 20, and https://api.github.com/graphql) should be moved to constants